src/panels/CreateClusterPanel.ts

```diff
@@ -162,11 +164,32 @@ async function createCluster(
         errorMessage: null,
     });
 
-    const clusterSpec = getManagedClusterSpec(location, name);
+    // kubernetes version is required to create a cluster via deployments
+    const kubernetesVersion = await containerServiceClient.managedClusters.listKubernetesVersions(location);
+    if (!kubernetesVersion || !kubernetesVersion.values || kubernetesVersion.values.length === 0 || !kubernetesVersion.values[0].version) {
+        window.showErrorMessage(`No Kubernetes versions available for location ${location}`);
+        webview.postProgressUpdate({
+            event: ProgressEventType.Failed,
+            operationDescription,
+            errorMessage: "No Kubernetes versions available for location",
+        });
+        return;
+    }
+
+    const clusterSpec: ClusterSpec = {
+        location,
+        name,
+        resourceGroupName: group.name,
+        subscriptionId: subscriptionId,
+        kubernetesVersion: kubernetesVersion.values[0].version, // selecting the latest version since versions come in descending order
+        username: username
+    };
+
+    const deploymentSpec = getManagedClusterSpec(clusterSpec, preset);
 
     try {
-        const poller = await containerServiceClient.managedClusters.beginCreateOrUpdate(group.name, name, clusterSpec);
-        poller.onProgress((state) => {
+        const poller = await resourceManagementClient.deployments.beginCreateOrUpdate(group.name, name, deploymentSpec);
```

Comment:
I was thinking it'd be a good idea to call the deployment validation API first - what do you think?

Reply:
I can think of two reasons for performing validation first:
1. I don't believe ARM template deployment is atomic. I.e. if there's an error during deployment your resources are not rolled back to their previous state. So, performing validation first should reduce the risk of users ending up with a partially deployed set of resources.
2. A good validation API should return any validation errors in a well-documented, strongly-typed object that's returned directly (as the return type) from the validation function.

However... Here's what I've learned today:

The validation API does _not_ work as I expected it to. It _looks_ as if it returns a [`DeploymentValidateResult`](https://learn.microsoft.com/en-us/javascript/api/@azure/arm-resources/deploymentvalidateresult?view=azure-node-latest) containing an `error` property if it encounters validation errors. But actually it throws an error, which you have to drill into to extract the real error, just like we have to for the `beginCreateOrUpdate` `catch` block.

Given the identical error thrown between `beginValidate` and `beginCreateOrUpdate`, it appears there's some overlap between the validation that happens in those functions. I guess ARM delegates to each service's RP to perform validation, and the AKS RP does the same validation in both cases. And, for the dev/test cluster at least, all the resources depend on the cluster, so if the cluster creation fails then other resources won't be created anyway. So I think my reason (1) for performing the validation first is moot.

@Tatsinnit - any thoughts about this? I _think_ all this means that an extra validation step is not strictly necessary, but maybe at least a comment in the code saying why we're not doing that would be good.

Reply:
I think we can agree we don't want validation api which is of not much use for this enhancement ? do you guys agree @Tatsinnit @peterbom ? Please confirm.

Reply:
I agree (for now).

Just want to note, though, in case we ever revisit this discussion: I certainly would consider adding it in the future, especially for the other presets where we're deploying more resources.